portfolio/src/main/java/com/google/sps/servlets/SentimentRecommenderServlet.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Book;
+import com.google.sps.data.BookDao;
+import com.google.sps.data.BookDaoDatastore;
+import com.google.sps.data.BookReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/sentimentRecommendationById")
+public class SentimentRecommenderServlet extends HttpServlet {
+  private Map<Book, List<Book>> hardCodedMappings;
+  private BookDao bookDao;
+  private static final List<String> ISBN_LIST = Arrays.asList("9780062310712", "9780062348654",
+      "9780062498557", "9780141363981", "9780316341646", "9780393609103", "9780525506348",
+      "9780525555377", "9780735212206", "9780735224315", "9780765387486", "9780812985405",
+      "9781250095251", "9781442468405", "9781473582231", "9781501160783", "9781501174827",
+      "9781524714703", "9781619634497", "9781681195803");
+
+  @Override
+  public void init() {
+    bookDao = new BookDaoDatastore();
+    makeHardCodedMappings();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    List<Book> bookList = bookDao.getBookList();
+    Book book = bookDao.getEntity(request.getParameter("isbn"));
+    String json = gson.toJson(hardCodedMappings.get(book));
```

Comment:
nit: no need for json variable. We can just combine this with the following line:
response.getWriter().println(gson.toJson(...))